### DISCUSSION TITLE
Collection of questions regarding a potential RBMK-Simulator

### DISCUSSION THREAD

**Discussion by LukeTech2006** [HbmMods/Hbm-s-Nuclear-Tech-GIT#847] [Q&A]
Hey there! I am a German IT student and am currently working on a stand-alone simulator for HBM's RBMK reactors.
I am missing a few puzzle pieces, and thats exactly why I opened this discussion thread.
One of my most prominent questions is "How does the heating work?", but this question is rather unprecise, so let me try to form a few better questions relating to this.

_1. "How much Water dissipates how much Heat per unit of time?"
2. "How much heat is dissipated through passive cooling per unit of time?"
3. "How much Coolant dissipates how much Heat per unit of time?"
4. "How do the different types of steam affect the aforementioned values?"_

I have tried to keep this as polite and precise is possible, so please ignore all grammatical errors. Also, if you have information, that you think is relevant for such a simulation, feel free to reach out to me via lukaskretschmar1@gmail.com or Discord (A3-5#4679).
Thank you for your answers!

Kind regards, Lukas

**Comment by HbmMods** [comment]
1. 1mB of water cools the boiler by 0.1°C, no matter what steam compression level is used. Per tick, the boiler will heat up water until either the water runs out or the boiler reaches the same temperature as the given steam type. Steam that doesn't fit into the output buffer will be voided, a full buffer doesn't stop the reactor from cooling.
2. By default, each part will cool by 1°C per tick.
3. In RBMK coolers, 1mB of cryogel will remove 1°C, keeping the cooler at a maximum of 750°C. If below that temperature, the cooler will do nothing. In the heat exchanger, using coolant is a bit more complicated, since the math has to match up properly with the TU (thermal unit) system, so it's necessary to convert from an RBMK's °C to TU. 1°C provides 2,000 TU for converting fluids.
Coolant conversions go as follows:
- 1mB oil + 10 TU = 1mB hot oil (350°C)
- 1mB cracked oil + 10 TU = 1mB hot cracked oil (350°C)
- 1mB coolant + 300 TU = 1mB hot coolant (600°C)
- 1mB MUG + 400 TU = 1mB hot MUG (500°C)
- 1mB water + 200 TU = 100mB steam (100°C)
- 10mB steam + 2 TU = 1mB  dense steam (300°C)
- 10mB dense steam + 18 TU = 1mB super dense steam (450°C)
- 10mB super dense steam + 120 TU = 1mB ultra dense steam (600°C)
The heat exchanger works in a similar fashion as the boiler, trying to heat up as much liquid as possible until the temperature of the heater matches the temperature of the output fluid. However, heating water and steam only has 50% efficiency which is applied at the very end of the calculation when heat is deducted, meaning that each tick, a water or steam cooled reactor using heat exchangers will only do a half-step towards the target temperature. Another difference to the boiler is that heat exchangers will never void fluid, if the buffer runs full it will stop cooling entirely.
4. Steam types only really exist to allow a higher through-put since the highest steam type is 1000x denser than regular steam. However, due to the increased temperature of higher pressure steam types, the reactor will also run hotter. For example, boilers with minimal compression will try to reach 100°C with the amount of water they have while UDS boilers will only cool down to 600°C.

**Reply by LukeTech2006** [reply]
Thank you. Another question that just popped into my head:

_"How does the heat transfer?"_

Is it just a simple function
`If otherColumn.Temperature < thisColumn.Temperature then
    otherColumn.Temperature -= 1
end`

Or is it more complicated?

**Reply by HbmMods** [reply]
Each tick, columns will detect other columns in cardinal directions, so up to 4 neighbors. Then the column will calculate the average temperature of all neighbors including itself, and then move each column's heat value towards that target average by a certain percentage. The default is 20%, so a column with 1000°C next to one that is 0°C will calculate an average of 500°C, then do a 20% step towards that average (resulting in 900°C and 100°C respectively). Note that the other column will also try to move heat in the same world tick, resulting in 820°C and 180°C respectively using the same rules as before.

ReaSim boilers also use the same rules for spreading water and steam between parts, but that transfer is always instant compared to heat's 20%/t spread.
